python/simplesauce/options.py

```diff
@@ -43,128 +95,87 @@ def _set_default_build_name(self):
         else:
             self.build = 'Build Time: {}'.format(datetime.utcnow())
 
-    def __init__(self, browserName=None, browserVersion=None, platformName=None, options={}):
-        self.options = {}
-        self.options['sauce:options'] = {}
-
-        self._test_name = 'test'
-        self._set_default_build_name()
-
-        self._set_defaults()
-        self._set_default_w3c_options()
-
-        if not any([browserName, browserVersion, platformName, options]):
-            return
-        elif not any([browserName, browserVersion, platformName]):
-            self.parse_options(options)
-            return
+    def _set_browser_name(self, browser_name):
+        if self.browser_name is None and browser_name is None:
+            self.set_capability('browserName', 'chrome')
+        elif browser_name and browser_name.lower() in browser_names.keys():
+            self.set_capability('browserName', browser_names[browser_name.lower()])
+        elif browser_name:
+            self.set_capability('browserName', browser_name.lower())
 
-        if browserName:
-            self.parse_browser_name(browserName)
-        if browserVersion:
-            self.parse_browser_version(browserVersion)
-        if platformName:
-            self.parse_platform_name(platformName)
+        self._set_platform_name()
 
-    @property
-    def name(self):
-        return self._test_name
+    def _set_platform_name(self):
+        if self.browser_name == 'safari' and self.browser_version == 'latest':
+            self.set_capability('platformName', 'macOS 10.14')
+        elif self.platform_name is None:
+            self.set_capability('platformName', 'Windows 10')
 
-    @name.setter
-    def name(self, name):
-        self._test_name = name
+    def __init__(self, browserName=None, **kwargs):
+        super(SauceOptions, self).__setattr__('options', {'sauce:options': {}})
+        super(SauceOptions, self).__setattr__('seleniumOptions', {})
 
-        self.options['sauce:options']['name'] = self._test_name
+        for key, value in kwargs.items():
+            if key is 'seleniumOptions':
+                if type(value) in option_types:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4711738)

Reply:
except, I legit do not want to use `isinstance()` since the whole point is to be able to do a lookup instead of a switch statement.